{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Finding TF motifs # "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### IMPORTANT: Please make sure that you are using the bash kernel to run this notebook. ###\n",
    "### IMPORTANT: Run the command below to git pull and make sure you are running the latest code!! ###\n",
    "#### (Do this at the beginning of every session) ###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No local changes to save\r\n",
      "Already up-to-date.\r\n"
     ]
    }
   ],
   "source": [
    "cd /srv/scratch/training_camp/tc2017/`whoami`/src/training_camp\n",
    "git stash \n",
    "git pull "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "### Set up variables storing the location of our data\n",
    "### The proper way to load your variables is with the ~/.bashrc command, but this is very slow in iPython \n",
    "export SUNETID=\"$(whoami)\"\n",
    "export WORK_DIR=\"/srv/scratch/training_camp/tc2017/${SUNETID}\"\n",
    "export DATA_DIR=\"${WORK_DIR}/data\"\n",
    "export FASTQ_DIR=\"${DATA_DIR}/fastq/\"\n",
    "export SRC_DIR=\"${WORK_DIR}/src/training_camp/src/\"\n",
    "\n",
    "export ANALYSIS_DIR=\"${WORK_DIR}/analysis/\"\n",
    "export TRIMMED_DIR=\"$ANALYSIS_DIR/trimmed\"\n",
    "export ALIGNMENT_DIR=\"$ANALYSIS_DIR/aligned/\"\n",
    "export TAGALIGN_DIR=\"$ANALYSIS_DIR/tagAlign/\"\n",
    "export PEAKS_DIR=\"$ANALYSIS_DIR/peaks/\"\n",
    "export SIGNAL_DIR=\"${ANALYSIS_DIR}signal/\"\n",
    "export DESEQ_DIR=\"${ANALYSIS_DIR}deseq/\"\n",
    "\n",
    "export FOLDCHANGE_DIR=\"${SIGNAL_DIR}foldChange/\"\n",
    "export COUNTS_DIR=\"${SIGNAL_DIR}counts/\"\n",
    "\n",
    "export YEAST_DIR=\"/srv/scratch/training_camp/saccer3/seq\"\n",
    "export YEAST_INDEX=\"/srv/scratch/training_camp/saccer3/bowtie2_index/saccer3\"\n",
    "export YEAST_CHR=\"/srv/scratch/training_camp/saccer3/sacCer3.chrom.sizes\"\n",
    "\n",
    "export TMP=\"${WORK_DIR}/tmp\"\n",
    "export TEMP=$TMP \n",
    "export TMPDIR=$TMP\n",
    "\n",
    "export RLIBS=$RLIBS:\"/usr/local/lib/R/site-library\"\n",
    "export MASTER_DATA=\"/srv/scratch/training_camp/data/tc2017\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this tutorial, we will focus on identifying motifs in the ATAC-seq peaks: \n",
    "![Analysis pipeline](part6.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will look for TF motifs in the open chromatin regions we have identified."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/srv/scratch/training_camp/tc2017/user1/analysis/motifs\r\n"
     ]
    }
   ],
   "source": [
    "export GENOME_FILE_DIR=${ANALYSIS_DIR}genomeFiles\n",
    "mkdir -p ${ANALYSIS_DIR}genomeFiles\n",
    "\n",
    "export MOTIF_DIR=${ANALYSIS_DIR}motifs\n",
    "mkdir -p ${ANALYSIS_DIR}motifs \n",
    "\n",
    "echo $MOTIF_DIR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scanning for motifs##"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will do our motif analysis using the software suite Homer.\n",
    "\n",
    "Homer looks at narrowPeak files located in your $PEAKS_DIR:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCD-0_6MNaCl-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCD-0_6MNaCl-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCD-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCD-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCE-0_6MNaCl-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCE-0_6MNaCl-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCE-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/WT-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCD-0_6MNaCl-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCD-0_6MNaCl-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCD-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCD-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCE-0_6MNaCl-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCE-0_6MNaCl-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCE-Rep1_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "/srv/scratch/training_camp/data/tc2017/peaks/cln3-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n"
     ]
    }
   ],
   "source": [
    "#files you can use for the analysis: \n",
    "ls $MASTER_DATA/peaks/*narrowPeak "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/srv/scratch/training_camp/tc2017/user1/analysis/genomeFiles\r\n",
      "/srv/scratch/training_camp/tc2017/user1/analysis/motifs\r\n",
      "\r\n",
      "\tPosition file = /srv/scratch/training_camp/data/tc2017/peaks/cln3-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "\tGenome = /srv/scratch/training_camp/saccer3/SacCer3_genome.fa\r\n",
      "\tOutput Directory = /srv/scratch/training_camp/tc2017/user1/analysis/motifs/cln3-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak\r\n",
      "\tUsing actual sizes of regions (-size given)\r\n",
      "\tFragment size set to given\r\n",
      "\tMotif length set at 8,\r\n",
      "\tUsing Custom Genome\r\n",
      "\tPeak/BED file conversion summary:\r\n",
      "\t\tBED/Header formatted lines: 4710\r\n",
      "\t\tpeakfile formatted lines: 0\r\n",
      "\r\n",
      "\tPeak File Statistics:\r\n",
      "\t\tTotal Peaks: 4710\r\n",
      "\t\tRedundant Peak IDs: 0\r\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\r\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\r\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\r\n",
      "\r\n",
      "\tPeak file looks good!\r\n",
      "\r\n",
      "\tBackground fragment size set to 574 (avg size of targets)\r\n",
      "ls: cannot access /srv/scratch/training_camp/tc2017/user1/analysis/motifs/SacCer3_genome.fa.*.cgbins: No such file or directory\r\n",
      "\tCould not find background files for 574 bp fragments\r\n",
      "\tBelow are the sizes that are already available prepared.\r\n",
      "\tHOMER will now create background files for 574 bp fragments\r\n",
      "\tTo CANCEL and rerun with a differet \"-size <#>\", hit <CTRL+C> now!\r\n",
      "\t\t5\r\n",
      "\t\t4\r\n",
      "\t\t3\r\n",
      "\t\t2\r\n",
      "\t\t1\r\n",
      "\tPreparsing genome for 574 bp fragments...(will probably take 1-5 min)\r\n",
      "\tUsing Custom Genome\r\n",
      "\tpreparse size set to 574\r\n",
      "\r\n",
      "\tNo reference file found - using random positions...\r\n",
      "\tOutput files will be placed in /srv/scratch/training_camp/tc2017/user1/analysis/motifs/SacCer3_genome.fa.574.* ...\r\n",
      "\tBy default, HOMER will set the the preparsed Directory to be group read/write\r\n",
      "\r\n",
      "\tChecking sequences from file: /srv/scratch/training_camp/saccer3/SacCer3_genome.fa\r\n",
      "\tExtracting sequences\r\n",
      "\r\n",
      "\tExtracting sequences from file: /srv/scratch/training_camp/saccer3/SacCer3_genome.fa\r\n",
      "\tLooking for peak sequences in a single file (/srv/scratch/training_camp/saccer3/SacCer3_genome.fa)\r\n",
      "\tExtracting 532 sequences from chrI\r\n",
      "\tExtracting 1576 sequences from chrII\r\n",
      "\tExtracting 678 sequences from chrIII\r\n",
      "\tExtracting 2910 sequences from chrIV\r\n",
      "\tExtracting 898 sequences from chrIX\r\n",
      "\tExtracting 248 sequences from chrM\r\n",
      "\tExtracting 1172 sequences from chrV\r\n",
      "\tExtracting 587 sequences from chrVI\r\n",
      "\tExtracting 2119 sequences from chrVII\r\n",
      "\tExtracting 1145 sequences from chrVIII\r\n",
      "\tExtracting 1512 sequences from chrX\r\n",
      "\tExtracting 1332 sequences from chrXI\r\n",
      "\tExtracting 2080 sequences from chrXII\r\n",
      "\tExtracting 1791 sequences from chrXIII\r\n",
      "\tExtracting 1539 sequences from chrXIV\r\n",
      "\tExtracting 2125 sequences from chrXV\r\n",
      "\tExtracting 1847 sequences from chrXVI\r\n",
      "\t\t6 of 22648 removed because they had >70.00% Ns (i.e. masked repeats)\r\n",
      "\r\n",
      "\tSequences processed:\r\n",
      "\t\tAuto detected maximum sequence length of 581 bp\r\n",
      "\t\t22642 total\r\n",
      "\r\n",
      "\tFrequency Bins: 0.01 0.02 0.03 0.04 0.05 0.06 0.08 0.1 0.12 0.14 0.16\r\n",
      "\tFreq\tBin\tCount\r\n",
      "\t0.01\t0\t137\r\n",
      "\t0.02\t1\t3084\r\n",
      "\t0.03\t2\t10442\r\n",
      "\t0.04\t3\t6098\r\n",
      "\t0.05\t4\t2221\r\n",
      "\t0.06\t5\t458\r\n",
      "\t0.08\t6\t179\r\n",
      "\t0.1\t7\t19\r\n",
      "\t0.12\t8\t4\r\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\r\n",
      "\tFreq\tBin\tCount\r\n",
      "\t0.2\t0\t113\r\n",
      "\t0.25\t1\t50\r\n",
      "\t0.3\t2\t237\r\n",
      "\t0.35\t3\t4185\r\n",
      "\t0.4\t4\t11184\r\n",
      "\t0.45\t5\t5605\r\n",
      "\t0.5\t6\t1109\r\n",
      "\t0.6\t7\t156\r\n",
      "\t0.7\t8\t3\r\n",
      "\r\n",
      "\r\n",
      "\tExtracting sequences from file: /srv/scratch/training_camp/saccer3/SacCer3_genome.fa\r\n",
      "\tLooking for peak sequences in a single file (/srv/scratch/training_camp/saccer3/SacCer3_genome.fa)\r\n",
      "\tExtracting 93 sequences from chrI\r\n",
      "\tExtracting 337 sequences from chrII\r\n",
      "\tExtracting 131 sequences from chrIII\r\n",
      "\tExtracting 567 sequences from chrIV\r\n",
      "\tExtracting 187 sequences from chrIX\r\n",
      "\tExtracting 29 sequences from chrM\r\n",
      "\tExtracting 247 sequences from chrV\r\n",
      "\tExtracting 110 sequences from chrVI\r\n",
      "\tExtracting 419 sequences from chrVII\r\n",
      "\tExtracting 219 sequences from chrVIII\r\n",
      "\tExtracting 295 sequences from chrX\r\n",
      "\tExtracting 250 sequences from chrXI\r\n",
      "\tExtracting 423 sequences from chrXII\r\n",
      "\tExtracting 344 sequences from chrXIII\r\n",
      "\tExtracting 319 sequences from chrXIV\r\n",
      "\tExtracting 401 sequences from chrXV\r\n",
      "\tExtracting 339 sequences from chrXVI\r\n",
      "\r\n",
      "\tNot removing redundant sequences\r\n",
      "\r\n",
      "\r\n",
      "\tSequences processed:\r\n",
      "\t\tAuto detected maximum sequence length of 643 bp\r\n",
      "\t\t4710 total\r\n",
      "\r\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\r\n",
      "\tFreq\tBin\tCount\r\n",
      "\t0.3\t2\t17\r\n",
      "\t0.35\t3\t244\r\n",
      "\t0.4\t4\t1407\r\n",
      "\t0.45\t5\t1935\r\n",
      "\t0.5\t6\t887\r\n",
      "\t0.6\t7\t220\r\n",
      "\r\n",
      "\tTotal sequences set to 50000\r\n",
      "\r\n",
      "\tChoosing background that matches in CpG/GC Content...\r\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\r\n",
      "\t2\t17\t237\t0.342\r\n",
      "\t3\t244\t4185\t0.278\r\n",
      "\t4\t1407\t11184\t0.600\r\n",
      "\t5\t1935\t5605\t1.647\r\n",
      "\t6\t887\t1109\t3.817\r\n",
      "\t7\t220\t156\t6.730\r\n",
      "\tAssembling sequence file...\r\n",
      "\tNormalizing lower order oligos using homer2\r\n",
      "\r\n",
      "\tReading input files...\r\n",
      "\t27186 total sequences read\r\n",
      "\tAutonormalization: 1-mers (4 total)\r\n",
      "\t\tA\t29.05%\t29.26%\t0.993\r\n",
      "\t\tC\t20.95%\t20.74%\t1.010\r\n",
      "\t\tG\t20.95%\t20.74%\t1.010\r\n",
      "\t\tT\t29.05%\t29.26%\t0.993\r\n",
      "\tAutonormalization: 2-mers (16 total)\r\n",
      "\t\tAA\t9.70%\t9.69%\t1.001\r\n",
      "\t\tCA\t6.39%\t6.76%\t0.946\r\n",
      "\t\tGA\t5.95%\t6.30%\t0.943\r\n",
      "\t\tTA\t7.02%\t6.50%\t1.079\r\n",
      "\t\tAC\t5.58%\t5.55%\t1.005\r\n",
      "\t\tCC\t4.59%\t4.51%\t1.019\r\n",
      "\t\tGC\t4.84%\t4.38%\t1.104\r\n",
      "\t\tTC\t5.95%\t6.30%\t0.943\r\n",
      "\t\tAG\t5.93%\t6.01%\t0.988\r\n",
      "\t\tCG\t4.03%\t3.46%\t1.166\r\n",
      "\t\tGG\t4.59%\t4.51%\t1.019\r\n",
      "\t\tTG\t6.39%\t6.76%\t0.946\r\n",
      "\t\tAT\t7.84%\t8.01%\t0.978\r\n",
      "\t\tCT\t5.93%\t6.01%\t0.988\r\n",
      "\t\tGT\t5.58%\t5.55%\t1.005\r\n",
      "\t\tTT\t9.70%\t9.69%\t1.001\r\n",
      "\tAutonormalization: 3-mers (64 total)\r\n",
      "\tNormalization weights can be found in file: /srv/scratch/training_camp/tc2017/user1/analysis/motifs/cln3-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak/seq.autonorm.tsv\r\n",
      "\tConverging on autonormalization solution:\r\n",
      "\t...............................................................................\r\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\r\n",
      "\t\tA\t29.05%\t29.28%\t0.992\r\n",
      "\t\tC\t20.95%\t20.72%\t1.011\r\n",
      "\t\tG\t20.95%\t20.72%\t1.011\r\n",
      "\t\tT\t29.05%\t29.28%\t0.992\r\n",
      "\tAutonormalization: 2-mers (16 total)\r\n",
      "\t\tAA\t9.70%\t9.76%\t0.993\r\n",
      "\t\tCA\t6.39%\t6.58%\t0.972\r\n",
      "\t\tGA\t5.95%\t6.13%\t0.970\r\n",
      "\t\tTA\t7.02%\t6.81%\t1.031\r\n",
      "\t\tAC\t5.58%\t5.53%\t1.009\r\n",
      "\t\tCC\t4.59%\t4.50%\t1.021\r\n",
      "\t\tGC\t4.84%\t4.57%\t1.059\r\n",
      "\t\tTC\t5.95%\t6.13%\t0.970\r\n",
      "\t\tAG\t5.93%\t5.96%\t0.996\r\n",
      "\t\tCG\t4.03%\t3.68%\t1.095\r\n",
      "\t\tGG\t4.59%\t4.50%\t1.021\r\n",
      "\t\tTG\t6.39%\t6.58%\t0.972\r\n",
      "\t\tAT\t7.84%\t8.03%\t0.976\r\n",
      "\t\tCT\t5.93%\t5.96%\t0.996\r\n",
      "\t\tGT\t5.58%\t5.53%\t1.009\r\n",
      "\t\tTT\t9.70%\t9.76%\t0.993\r\n",
      "\tAutonormalization: 3-mers (64 total)\r\n",
      "\tFinished preparing sequence/group files\r\n",
      "\r\n",
      "\t----------------------------------------------------------\r\n",
      "\tKnown motif enrichment\r\n",
      "\r\n",
      "\tReading input files...\r\n",
      "\t27186 total sequences read\r\n",
      "\t387 motifs loaded\r\n",
      "\tCache length = 11180\r\n",
      "\tUsing binomial scoring\r\n",
      "\tChecking enrichment of 387 motif(s)\r\n",
      "\t|0%                                    50%                                  100%|\r\n",
      "\t=================================================================================\r\n",
      "\tPreparing HTML output with sequence logos...\r\n",
      "\t\t1 of 387 (1e-138) ABF1/SacCer-Promoters/Homer\r\n",
      "\t\t2 of 387 (1e-128) REB1/SacCer-Promoters/Homer\r\n",
      "\t\t3 of 387 (1e-65) RLR1?/SacCer-Promoters/Homer\r\n",
      "\t\t4 of 387 (1e-48) DPL-1(E2F)/cElegans-Adult-ChIP-Seq(modEncode)/Homer\r\n",
      "\t\t5 of 387 (1e-44) c-Myc(bHLH)/LNCAP-cMyc-ChIP-Seq(Unpublished)/Homer\r\n",
      "\t\t6 of 387 (1e-43) SeqBias: CG-repeat\r\n",
      "\t\t7 of 387 (1e-43) Cbf1(bHLH)/Yeast-Cbf1-ChIP-Seq(GSE29506)/Homer\r\n",
      "\t\t8 of 387 (1e-36) bHLHE40(bHLH)/HepG2-BHLHE40-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t9 of 387 (1e-36) Usf2(bHLH)/C2C12-Usf2-ChIP-Seq(GSE36030)/Homer\r\n",
      "\t\t10 of 387 (1e-29) n-Myc(bHLH)/mES-nMyc-ChIP-Seq(GSE11431)/Homer\r\n",
      "\t\t11 of 387 (1e-28) USF1(bHLH)/GM12878-Usf1-ChIP-Seq(GSE32465)/Homer\r\n",
      "\t\t12 of 387 (1e-27) CLOCK(bHLH)/Liver-Clock-ChIP-Seq(GSE39860)/Homer\r\n",
      "\t\t13 of 387 (1e-25) Pho2(bHLH)/Yeast-Pho2-ChIP-Seq(GSE29506)/Homer\r\n",
      "\t\t14 of 387 (1e-23) E-box(bHLH)/Promoter/Homer\r\n",
      "\t\t15 of 387 (1e-22) E-box/Arabidopsis-Promoters/Homer\r\n",
      "\t\t16 of 387 (1e-22) FHY3(FAR1)/Arabidopsis-FHY3-ChIP-Seq(GSE30711)/Homer\r\n",
      "\t\t17 of 387 (1e-22) PIF5ox(bHLH)/Arabidopsis-PIF5ox-ChIP-Seq(GSE35062)/Homer\r\n",
      "\t\t18 of 387 (1e-21) SFP1/SacCer-Promoters/Homer\r\n",
      "\t\t19 of 387 (1e-20) Max(bHLH)/K562-Max-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t20 of 387 (1e-17) PIF4(bHLH)/Seedling-PIF4-ChIP-Seq(GSE35315)/Homer\r\n",
      "\t\t21 of 387 (1e-17) SPCH(bHLH)/Seedling-SPCH-ChIP-Seq(GSE57497)/Homer\r\n",
      "\t\t22 of 387 (1e-16) IBL1(bHLH)/Seedling-IBL1-ChIP-Seq(GSE51120)/Homer\r\n",
      "\t\t23 of 387 (1e-16) MITF(bHLH)/MastCells-MITF-ChIP-Seq(GSE48085)/Homer\r\n",
      "\t\t24 of 387 (1e-15) NPAS2(bHLH)/Liver-NPAS2-ChIP-Seq(GSE39860)/Homer\r\n",
      "\t\t25 of 387 (1e-14) TATA-box/SacCer-Promoters/Homer\r\n",
      "\t\t26 of 387 (1e-14) E2F4(E2F)/K562-E2F4-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t27 of 387 (1e-12) TOD6?/SacCer-Promoters/Homer\r\n",
      "\t\t28 of 387 (1e-12) KLF14(Zf)/HEK293-KLF14.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t29 of 387 (1e-11) HIF-1a(bHLH)/MCF7-HIF1a-ChIP-Seq(GSE28352)/Homer\r\n",
      "\t\t30 of 387 (1e-10) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\r\n",
      "\t\t31 of 387 (1e-9) HIF-1b(HLH)/T47D-HIF1b-ChIP-Seq(GSE59937)/Homer\r\n",
      "\t\t32 of 387 (1e-9) CRE(bZIP)/Promoter/Homer\r\n",
      "\t\t33 of 387 (1e-9) Arnt:Ahr(bHLH)/MCF7-Arnt-ChIP-Seq(Lo_et_al.)/Homer\r\n",
      "\t\t34 of 387 (1e-9) AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\r\n",
      "\t\t35 of 387 (1e-8) BMAL1(bHLH)/Liver-Bmal1-ChIP-Seq(GSE39860)/Homer\r\n",
      "\t\t36 of 387 (1e-7) HIF2a(bHLH)/785_O-HIF2a-ChIP-Seq(GSE34871)/Homer\r\n",
      "\t\t37 of 387 (1e-7) Maz(Zf)/HepG2-Maz-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t38 of 387 (1e-7) Egr2(Zf)/Thymocytes-Egr2-ChIP-Seq(GSE34254)/Homer\r\n",
      "\t\t39 of 387 (1e-7) Pho4(bHLH)/Yeast-Pho4-ChIP-Seq(GSE29506)/Homer\r\n",
      "\t\t40 of 387 (1e-7) Fra1(bZIP)/BT549-Fra1-ChIP-Seq(GSE46166)/Homer\r\n",
      "\t\t41 of 387 (1e-6) c-Myc(bHLH)/mES-cMyc-ChIP-Seq(GSE11431)/Homer\r\n",
      "\t\t42 of 387 (1e-6) ZBTB33(Zf)/GM12878-ZBTB33-ChIP-Seq(GSE32465)/Homer\r\n",
      "\t\t43 of 387 (1e-6) Egr1(Zf)/K562-Egr1-ChIP-Seq(GSE32465)/Homer\r\n",
      "\t\t44 of 387 (1e-5) GLI3(Zf)/Limb-GLI3-ChIP-Chip(GSE11077)/Homer\r\n",
      "\t\t45 of 387 (1e-5) Atf3(bZIP)/GBM-ATF3-ChIP-Seq(GSE33912)/Homer\r\n",
      "\t\t46 of 387 (1e-5) Zic(Zf)/Cerebellum-ZIC1.2-ChIP-Seq(GSE60731)/Homer\r\n",
      "\t\t47 of 387 (1e-5) Elk4(ETS)/Hela-Elk4-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t48 of 387 (1e-5) TATA-box/Drosophila-Promoters/Homer\r\n",
      "\t\t49 of 387 (1e-4) p53(p53)/Saos-p53-ChIP-Seq(GSE15780)/Homer\r\n",
      "\t\t50 of 387 (1e-4) p53(p53)/Saos-p53-ChIP-Seq/Homer\r\n",
      "\t\t51 of 387 (1e-4) Pax8(Paired,Homeobox)/Thyroid-Pax8-ChIP-Seq(GSE26938)/Homer\r\n",
      "\t\t52 of 387 (1e-4) KLF5(Zf)/LoVo-KLF5-ChIP-Seq(GSE49402)/Homer\r\n",
      "\t\t53 of 387 (1e-4) ZNF519(Zf)/HEK293-ZNF519.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t54 of 387 (1e-4) GAGA-repeat/Arabidopsis-Promoters/Homer\r\n",
      "\t\t55 of 387 (1e-3) E2F7(E2F)/Hela-E2F7-ChIP-Seq(GSE32673)/Homer\r\n",
      "\t\t56 of 387 (1e-3) ZNF467(Zf)/HEK293-ZNF467.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t57 of 387 (1e-3) Elk1(ETS)/Hela-Elk1-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t58 of 387 (1e-3) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\r\n",
      "\t\t59 of 387 (1e-3) PAX5(Paired,Homeobox)/GM12878-PAX5-ChIP-Seq(GSE32465)/Homer\r\n",
      "\t\t60 of 387 (1e-3) NRF1(NRF)/MCF7-NRF1-ChIP-Seq(Unpublished)/Homer\r\n",
      "\t\t61 of 387 (1e-3) SpiB(ETS)/OCILY3-SPIB-ChIP-Seq(GSE56857)/Homer\r\n",
      "\t\t62 of 387 (1e-3) KLF10(Zf)/HEK293-KLF10.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t63 of 387 (1e-3) ZSCAN22(Zf)/HEK293-ZSCAN22.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t64 of 387 (1e-3) NRF(NRF)/Promoter/Homer\r\n",
      "\t\t65 of 387 (1e-3) Dorsal(RHD)/Embryo-dl-ChIP-Seq(GSE65441)/Homer\r\n",
      "\t\t66 of 387 (1e-3) E2F1(E2F)/Hela-E2F1-ChIP-Seq(GSE22478)/Homer\r\n",
      "\t\t67 of 387 (1e-2) SUT1?/SacCer-Promoters/Homer\r\n",
      "\t\t68 of 387 (1e-2) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\r\n",
      "\t\t69 of 387 (1e-2) E2F(E2F)/Hela-CellCycle-Expression/Homer\r\n",
      "\t\t70 of 387 (1e-2) ARE(NR)/LNCAP-AR-ChIP-Seq(GSE27824)/Homer\r\n",
      "\t\t71 of 387 (1e-2) BORIS(Zf)/K562-CTCFL-ChIP-Seq(GSE32465)/Homer\r\n",
      "\t\t72 of 387 (1e-2) Klf4(Zf)/mES-Klf4-ChIP-Seq(GSE11431)/Homer\r\n",
      "\t\t73 of 387 (1e-2) ZNF322(Zf)/HEK293-ZNF322.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t74 of 387 (1e-2) GEI-11(Myb?)/cElegans-L4-GEI11-ChIP-Seq(modEncode)/Homer\r\n",
      "\t\t75 of 387 (1e-2) MafK(bZIP)/C2C12-MafK-ChIP-Seq(GSE36030)/Homer\r\n",
      "\t\t76 of 387 (1e-2) Trl(Zf)/S2-GAGAfactor-ChIP-Seq(GSE40646)/Homer\r\n",
      "\t\t77 of 387 (1e-2) RXR(NR),DR1/3T3L1-RXR-ChIP-Seq(GSE13511)/Homer\r\n",
      "\t\t78 of 387 (1e-2) Fli1(ETS)/CD8-FLI-ChIP-Seq(GSE20898)/Homer\r\n",
      "\t\t79 of 387 (1e-2) ZNF692(Zf)/HEK293-ZNF692.GFP-ChIP-Seq(GSE58341)/Homer\r\n",
      "\t\t80 of 387 (1e-2) Sp1(Zf)/Promoter/Homer\r\n",
      "\t\t81 of 387 (1e-2) Oct4:Sox17(POU,Homeobox,HMG)/F9-Sox17-ChIP-Seq(GSE44553)/Homer\r\n",
      "\t\t82 of 387 (1e-2) PPARE(NR),DR1/3T3L1-Pparg-ChIP-Seq(GSE13511)/Homer\r\n",
      "\t----------------------------------------------------------\r\n",
      "\tDe novo motif finding (HOMER)\r\n",
      "\r\n",
      "\tScanning input files...\r\n",
      "\tParsing sequences...\r\n",
      "\t|0%                                   50%                                  100%|\r\n",
      "\t================================================================================\r\n",
      "\tTotal number of Oligos: 32896\r\n",
      "\tAutoadjustment for sequence coverage in background: 1.39x\r\n",
      "\r\n",
      "\tOligos: 32896 of 34497 max\r\n",
      "\tTree  : 67296 of 172485 max\r\n",
      "\tOptimizing memory usage...\r\n",
      "\tCache length = 11180\r\n",
      "\tUsing binomial scoring\r\n",
      "\r\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\r\n",
      "\r\n",
      "\tScreening oligos 32896 (allowing 0 mismatches):\r\n",
      "\t|0%                                   50%                                  100%|\r\n",
      "\t================================================================================\r\n",
      "\t\t0.00% skipped, 100.00% checked (32896 of 32896), of those checked:\r\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\r\n",
      "\r\n",
      "\tScreening oligos 32896 (allowing 1 mismatches):\r\n",
      "\t|0%                                   50%                                  100%|\r\n",
      "\t================================================================================\r\n",
      "\t\t0.00% skipped, 100.00% checked (32896 of 32896), of those checked:\r\n",
      "\t\t0.00% not in target, 44.33% increased p-value, 44.42% high p-value\r\n",
      "\r\n",
      "\tScreening oligos 32896 (allowing 2 mismatches):\r\n",
      "\t|0%                                   50%                                  100%|\r\n",
      "\t================================================================================\r\n",
      "\t\t56.41% skipped, 43.59% checked (14341 of 32896), of those checked:\r\n",
      "\t\t0.00% not in target, 25.26% increased p-value, 0.00% high p-value\r\n",
      "\tReading input files...\r\n",
      "\t27186 total sequences read\r\n",
      "\tCache length = 11180\r\n",
      "\tUsing binomial scoring\r\n",
      "\r\n",
      "\tLocal Optimization Phase:\r\n",
      "\t1 of 25 Initial Sequence: AAAAAAAA... (-305.686)\r\n",
      "\t\tRound 1: -381.10 RRAAAAAA T:8212.2(82.51%),B:27169.4(64.40%),P:1e-165\r\n",
      "\t\tRound 2: -381.10 RRAAAAAA T:8212.2(82.51%),B:27169.4(64.40%),P:1e-165\r\n",
      "\t\t=Final=: -109.62 RRAAAAAA T:1915.0(40.66%),B:5812.6(30.62%),P:1e-47\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t2 of 25 Initial Sequence: GCGCGCGG... (-213.238)\r\n",
      "\t\tRound 1: -273.42 GCGCGCGG T:3882.1(56.15%),B:13157.7(39.36%),P:1e-118\r\n",
      "\t\tRound 2: -281.27 GCGCGSNG T:3689.4(54.31%),B:12311.5(37.38%),P:1e-122\r\n",
      "\t\tRound 3: -336.40 GCGCGBNG T:3773.3(55.12%),B:11989.1(36.60%),P:1e-146\r\n",
      "\t\tRound 4: -359.51 GCGCGBNG T:4105.1(58.17%),B:12956.6(38.89%),P:1e-156\r\n",
      "\t\tRound 5: -359.51 GCGCGBNG T:4105.1(58.17%),B:12956.6(38.89%),P:1e-156\r\n",
      "\t\t=Final=: -209.29 GCGCGBNG T:1772.0(37.62%),B:4614.7(24.31%),P:1e-90\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t3 of 25 Initial Sequence: TTACCCGG... (-173.811)\r\n",
      "\t\tRound 1: -313.13 TTACCCGG T:1361.0(25.08%),B:3329.3(11.89%),P:1e-135\r\n",
      "\t\tRound 2: -313.13 TTACCCGG T:1361.0(25.08%),B:3329.3(11.89%),P:1e-135\r\n",
      "\t\t=Final=: -305.10 TTACCCGG T:874.0(18.56%),B:1430.0(7.53%),P:1e-132\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t4 of 25 Initial Sequence: TATATACG... (-83.251)\r\n",
      "\t\tRound 1: -122.47 TATATACG T:2836.8(45.24%),B:11064.3(34.34%),P:1e-53\r\n",
      "\t\tRound 2: -127.47 TATRTASG T:2644.5(42.96%),B:10139.9(31.99%),P:1e-55\r\n",
      "\t\tRound 3: -127.47 TATRTASG T:2644.5(42.96%),B:10139.9(31.99%),P:1e-55\r\n",
      "\t\t=Final=: -67.42 TATRTASG T:1929.0(40.96%),B:6274.3(33.05%),P:1e-29\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t5 of 25 Initial Sequence: GTCACGTG... (-72.010)\r\n",
      "\t\tRound 1: -130.94 GTCACGTG T:703.9(13.87%),B:1955.0(7.16%),P:1e-56\r\n",
      "\t\tRound 2: -130.94 GTCACGTG T:703.9(13.87%),B:1955.0(7.16%),P:1e-56\r\n",
      "\t\t=Final=: -124.27 GTCACGTG T:331.0(7.03%),B:504.8(2.66%),P:1e-53\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t6 of 25 Initial Sequence: CCCGCCCG... (-65.796)\r\n",
      "\t\tRound 1: -113.10 CCCGCCCG T:2857.6(45.48%),B:11329.1(34.99%),P:1e-49\r\n",
      "\t\tRound 2: -120.52 CCCGCCCG T:2556.6(41.88%),B:9863.4(31.27%),P:1e-52\r\n",
      "\t\tRound 3: -120.52 CCCGCCCG T:2556.6(41.88%),B:9863.4(31.27%),P:1e-52\r\n",
      "\t\t=Final=: -90.14 CCCGCCCG T:1385.0(29.41%),B:4030.0(21.23%),P:1e-39\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t7 of 25 Initial Sequence: CAAAAAAA... (-48.940)\r\n",
      "\t\tRound 1: -77.79 CAAAAAAA T:2393.7(39.84%),B:9915.9(31.40%),P:1e-33\r\n",
      "\t\tRound 2: -77.79 CAAAAAAA T:2393.7(39.84%),B:9915.9(31.40%),P:1e-33\r\n",
      "\t\t=Final=: -47.36 CAAAAAAA T:867.0(18.41%),B:2563.9(13.51%),P:1e-20\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t8 of 25 Initial Sequence: CCCACACA... (-38.040)\r\n",
      "\t\tRound 1: -76.60 CCCACACA T:2700.7(43.63%),B:11363.7(35.08%),P:1e-33\r\n",
      "\t\tRound 2: -93.10 CMCACACW T:1950.6(33.90%),B:7614.6(25.13%),P:1e-40\r\n",
      "\t\tRound 3: -93.10 CMCACACW T:1950.6(33.90%),B:7614.6(25.13%),P:1e-40\r\n",
      "\t\t=Final=: -84.22 CMCACACW T:1313.0(27.88%),B:3821.9(20.13%),P:1e-36\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t9 of 25 Initial Sequence: CGGTTCGA... (-36.474)\r\n",
      "\t\tRound 1: -88.44 GGGTTCGA T:308.5(6.33%),B:724.8(2.71%),P:1e-38\r\n",
      "\t\tRound 2: -88.44 GGGTTCGA T:308.5(6.33%),B:724.8(2.71%),P:1e-38\r\n",
      "\t\t=Final=: -90.13 GGGTTCGA T:437.0(9.28%),B:892.8(4.70%),P:1e-39\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t10 of 25 Initial Sequence: CCTTAACC... (-36.005)\r\n",
      "\t\tRound 1: -36.01 CCTTAACC T:96.2(2.02%),B:203.0(0.77%),P:1e-15\r\n",
      "\t\tRound 2: -36.01 CCTTAACC T:96.2(2.02%),B:203.0(0.77%),P:1e-15\r\n",
      "\t\t=Final=: -42.79 CCTTAACC T:125.0(2.65%),B:202.4(1.07%),P:1e-18\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t11 of 25 Initial Sequence: TTACCCTA... (-31.489)\r\n",
      "\t\tRound 1: -76.39 TTACCCTA T:686.1(13.56%),B:2275.9(8.29%),P:1e-33\r\n",
      "\t\tRound 2: -76.39 TTACCCTA T:686.1(13.56%),B:2275.9(8.29%),P:1e-33\r\n",
      "\t\t=Final=: -50.57 TTACCCTA T:648.0(13.76%),B:1777.2(9.36%),P:1e-21\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t12 of 25 Initial Sequence: TGGCCGAG... (-31.407)\r\n",
      "\t\tRound 1: -52.23 CGGCCGAG T:559.1(11.19%),B:1960.4(7.18%),P:1e-22\r\n",
      "\t\tRound 2: -60.56 MGGCCGAG T:351.7(7.18%),B:1027.7(3.83%),P:1e-26\r\n",
      "\t\tRound 3: -60.56 MGGCCGAG T:351.7(7.18%),B:1027.7(3.83%),P:1e-26\r\n",
      "\t\t=Final=: -83.24 MGGCCGAG T:451.0(9.58%),B:961.4(5.06%),P:1e-36\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t13 of 25 Initial Sequence: ACTGCGCT... (-28.011)\r\n",
      "\t\tRound 1: -46.35 ACTGCGCT T:2066.1(35.51%),B:9089.7(29.22%),P:1e-20\r\n",
      "\t\tRound 2: -64.46 ASTGCGCW T:2673.0(43.31%),B:11521.5(35.47%),P:1e-27\r\n",
      "\t\tRound 3: -71.19 ASTGCGCA T:2713.2(43.79%),B:11548.9(35.53%),P:1e-30\r\n",
      "\t\tRound 4: -74.32 ASTGYGCW T:2074.2(35.62%),B:8509.0(27.63%),P:1e-32\r\n",
      "\t\tRound 5: -74.32 ASTGYGCW T:2074.2(35.62%),B:8509.0(27.63%),P:1e-32\r\n",
      "\t\t=Final=: -36.23 ASTGYGCW T:902.0(19.15%),B:2804.7(14.78%),P:1e-15\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t14 of 25 Initial Sequence: CTTTTTCC... (-28.301)\r\n",
      "\t\tRound 1: -50.98 CTTTTTCC T:1199.3(22.48%),B:4872.8(16.91%),P:1e-22\r\n",
      "\t\tRound 2: -50.98 CTTTTTCC T:1199.3(22.48%),B:4872.8(16.91%),P:1e-22\r\n",
      "\t\t=Final=: -46.16 CTTTTTCC T:648.0(13.76%),B:1812.6(9.55%),P:1e-20\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t15 of 25 Initial Sequence: ATGCACGG... (-23.016)\r\n",
      "\t\tRound 1: -39.36 ATGCACGG T:225.4(4.67%),B:662.1(2.49%),P:1e-17\r\n",
      "\t\tRound 2: -39.36 ATGCACGG T:225.4(4.67%),B:662.1(2.49%),P:1e-17\r\n",
      "\t\t=Final=: -71.10 ATGCACGG T:286.0(6.07%),B:538.0(2.83%),P:1e-30\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t16 of 25 Initial Sequence: GAAAAATT... (-22.969)\r\n",
      "\t\tRound 1: -35.69 GAAAAATT T:1124.6(21.23%),B:4802.6(16.69%),P:1e-15\r\n",
      "\t\tRound 2: -35.69 GAAAAATT T:1124.6(21.23%),B:4802.6(16.69%),P:1e-15\r\n",
      "\t\t=Final=: -19.38 GAAAAATT T:547.0(11.61%),B:1726.1(9.09%),P:1e-8\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t17 of 25 Initial Sequence: CCCTTCCC... (-21.354)\r\n",
      "\t\tRound 1: -36.21 CCCTTCCC T:2900.7(45.98%),B:13481.2(40.10%),P:1e-15\r\n",
      "\t\tRound 2: -36.21 CCCTTCCC T:2900.7(45.98%),B:13481.2(40.10%),P:1e-15\r\n",
      "\t\t=Final=: -37.72 CCCTTCCC T:1077.0(22.87%),B:3425.9(18.05%),P:1e-16\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t18 of 25 Initial Sequence: GGCATCGC... (-21.210)\r\n",
      "\t\tRound 1: -26.18 SKCATCGC T:275.7(5.67%),B:978.5(3.65%),P:1e-11\r\n",
      "\t\tRound 2: -27.17 GKGATCGC T:352.4(7.20%),B:1311.9(4.86%),P:1e-11\r\n",
      "\t\tRound 3: -34.17 CGCATCGC T:220.7(4.56%),B:675.4(2.53%),P:1e-14\r\n",
      "\t\tRound 4: -34.17 CGCATCGC T:220.7(4.56%),B:675.4(2.53%),P:1e-14\r\n",
      "\t\t=Final=: -47.05 CGCATCGC T:292.0(6.20%),B:652.8(3.44%),P:1e-20\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t19 of 25 Initial Sequence: TTTGCCAC... (-20.977)\r\n",
      "\t\tRound 1: -20.98 TTTGCCAC T:155.6(3.24%),B:505.5(1.90%),P:1e-9\r\n",
      "\t\tRound 2: -20.98 TTTGCCAC T:155.6(3.24%),B:505.5(1.90%),P:1e-9\r\n",
      "\t\t=Final=: -29.04 TTTGCCAC T:211.0(4.48%),B:498.3(2.63%),P:1e-12\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t20 of 25 Initial Sequence: TACGTAAT... (-18.991)\r\n",
      "\t\tRound 1: -47.25 TACGTAAT T:949.6(18.25%),B:3773.7(13.36%),P:1e-20\r\n",
      "\t\tRound 2: -47.25 TACGTAAT T:949.6(18.25%),B:3773.7(13.36%),P:1e-20\r\n",
      "\t\t=Final=: -62.04 TACGTAAT T:454.0(9.64%),B:1071.2(5.64%),P:1e-26\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t21 of 25 Initial Sequence: TTTTTTTA... (-18.215)\r\n",
      "\t\tRound 1: -25.62 TTTTTTTR T:1805.6(31.84%),B:8411.6(27.37%),P:1e-11\r\n",
      "\t\tRound 2: -25.62 TTTTTTTR T:1805.6(31.84%),B:8411.6(27.37%),P:1e-11\r\n",
      "\t\t=Final=: -22.43 TTTTTTTR T:1075.0(22.82%),B:3633.6(19.14%),P:1e-9\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t22 of 25 Initial Sequence: CGCGCTTC... (-18.184)\r\n",
      "\t\tRound 1: -3.68 CGCGCTTC T:87.8(1.83%),B:388.0(1.46%),P:1e-1\r\n",
      "\t\tRound 2: -14.67 CGCKCTTC T:1884.0(32.96%),B:9245.0(29.63%),P:1e-6\r\n",
      "\t\tRound 3: -27.91 TKCGCTCY T:987.1(18.91%),B:4301.0(15.08%),P:1e-12\r\n",
      "\t\tRound 4: -27.91 TKCGCTCY T:987.1(18.91%),B:4301.0(15.08%),P:1e-12\r\n",
      "\t\t=Final=: -41.83 TKCGCTCY T:1246.0(26.45%),B:3999.3(21.07%),P:1e-18\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t23 of 25 Initial Sequence: CCGGCTAG... (-15.541)\r\n",
      "\t\tRound 1: -29.43 CCGGCTAG T:1051.6(20.00%),B:4581.7(15.98%),P:1e-12\r\n",
      "\t\tRound 2: -33.51 CCGGCTAG T:602.8(12.00%),B:2372.4(8.62%),P:1e-14\r\n",
      "\t\tRound 3: -33.51 CCGGCTAG T:602.8(12.00%),B:2372.4(8.62%),P:1e-14\r\n",
      "\t\t=Final=: -39.76 CCGGCTAG T:726.0(15.41%),B:2139.3(11.27%),P:1e-17\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t24 of 25 Initial Sequence: TAAGGCGC... (-15.339)\r\n",
      "\t\tRound 1: -20.38 TAAGGCGC T:122.2(2.56%),B:373.7(1.41%),P:1e-8\r\n",
      "\t\tRound 2: -21.99 TAAGGCKC T:252.1(5.21%),B:918.0(3.43%),P:1e-9\r\n",
      "\t\tRound 3: -26.17 TACGGCGC T:310.0(6.35%),B:1129.0(4.20%),P:1e-11\r\n",
      "\t\tRound 4: -26.17 TACGGCGC T:310.0(6.35%),B:1129.0(4.20%),P:1e-11\r\n",
      "\t\t=Final=: -41.87 TACGGCGC T:417.0(8.85%),B:1073.6(5.66%),P:1e-18\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\t25 of 25 Initial Sequence: GCGCGTCC... (-13.647)\r\n",
      "\t\tRound 1: -9.24 NCGCGTCN T:115.1(2.41%),B:439.7(1.66%),P:1e-4\r\n",
      "\t\tRound 2: -15.26 TCGCGTCA T:643.1(12.76%),B:2901.1(10.44%),P:1e-6\r\n",
      "\t\tRound 3: -19.63 TCGHGTCA T:223.3(4.62%),B:812.6(3.04%),P:1e-8\r\n",
      "\t\tRound 4: -19.63 TCGHGTCA T:223.3(4.62%),B:812.6(3.04%),P:1e-8\r\n",
      "\t\t=Final=: -32.06 TCGHGTCA T:310.0(6.58%),B:792.5(4.18%),P:1e-13\r\n",
      "\t\tPerforming exhaustive masking of motif...\r\n",
      "\t\tReprioritizing potential motifs...\r\n",
      "\r\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\r\n",
      "\tReading input files...\r\n",
      "\t27186 total sequences read\r\n",
      "\tCache length = 11180\r\n",
      "\tUsing binomial scoring\r\n",
      "\tChecking enrichment of 25 motif(s)\r\n",
      "\t|0%                                    50%                                  100%|\r\n",
      "\t=================================================================================\r\n",
      "\tOutput in file: /srv/scratch/training_camp/tc2017/user1/analysis/motifs/cln3-SCE-Rep2_R1_001.trimmed.nodup_peaks.narrowPeak/homerMotifs.motifs8\r\n",
      "\r\n",
      "\t(Motifs in homer2 format)\r\n",
      "\tDetermining similar motifs... 25 reduced to 15 motifs\r\n",
      "\tOutputing HTML and sequence logos for motif comparison...\r\n",
      "\tChecking de novo motifs against known motifs...\r\n",
      "\tFormatting HTML page...\r\n",
      "\t\t1 of 15 (1e-137) similar to REB1(MacIsaac)/Yeast(0.993)\r\n",
      "\t\t2 of 15 (1e-90) similar to RSC3/MA0374.1/Jaspar(0.953)\r\n",
      "\t\t3 of 15 (1e-54) similar to TYE7(MacIsaac)/Yeast(0.994)\r\n",
      "\t\t4 of 15 (1e-47) similar to hb/dmmpmm(SeSiMCMC)/fly(0.882)\r\n",
      "\t\t5 of 15 (1e-45) similar to btd/dmmpmm(Noyes)/fly(0.797)\r\n",
      "\t\t6 of 15 (1e-42) similar to OPI1/Literature(Harbison)/Yeast(0.875)\r\n",
      "\t\t7 of 15 (1e-38) similar to MET31(MacIsaac)/Yeast(0.789)\r\n",
      "\t\t8 of 15 (1e-33) similar to FHL1(MacIsaac)/Yeast(0.812)\r\n",
      "\t\t9 of 15 (1e-32) similar to YAP3/MA0416.1/Jaspar(0.947)\r\n",
      "\t\t10 of 15 (1e-31) similar to IME1(MacIsaac)/Yeast(0.841)\r\n",
      "\t\t11 of 15 (1e-25) similar to CHA4(MacIsaac)/Yeast(0.856)\r\n",
      "\t\t12 of 15 (1e-20) similar to ZNF467(Zf)/HEK293-ZNF467.GFP-ChIP-Seq(GSE58341)/Homer(0.818)\r\n",
      "\t\t13 of 15 (1e-18) similar to STP2/MA0395.1/Jaspar(0.804)\r\n",
      "\t\t14 of 15 (1e-16) similar to SWI6(MacIsaac)/Yeast(0.839)\r\n",
      "\t\t15 of 15 (1e-13) similar to RPN4/RPN4_H2O2Lo/[](Harbison)/Yeast(0.941)\r\n",
      "\tJob finished - if results look good, please send beer to ..\r\n",
      "\r\n",
      "\tCleaning up tmp files...\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "${SRC_DIR}/find_motifs.sh  <peak file of interest>\n",
    " #add your peak file of interest here "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For advanced or ambitious students: \n",
    "Use the HOMER software suite to perform discriminative discovery."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
